File: 509-fibonacci-number/509-fibonacci-number.kt
import kotlin.math.*
class Solution {
    fun fib(n: Int): Int {
        val sqrt5: Double = sqrt(5.toDouble());
        val phi: Double = (1 + sqrt5) / 2;
        val res: Double = round(phi.pow(n.toDouble()) / sqrt5);
        return res.toInt();
    }
}

File: 509-fibonacci-number/509-fibonacci-number.cs
public class Solution {
    public int Fib(int n) {
        // use explicit formula
        double phi = (1 + Math.Sqrt(5)) / 2;
        double res = Math.Round(Math.Pow(phi, n) / Math.Sqrt(5));
        return (int) res;
    }
}